### PR TITLE
MTU negotiation

### PR DESCRIPTION
**Pull request by julianbermudez**
This pull request add a `mtu` member to `CLIENT_Representation` in order to negotiate the MTU with the `Agent`.

### CODE REVIEW

File: src/c/core/serialization/xrce_protocol_internal.h [hunk 1]
```diff
@@ -201,7 +201,7 @@ typedef struct Property
 typedef struct PropertySeq
 {
     uint32_t size;
-    Property* data;
+    Property data[UXR_PROPERTY_SEQUENCE_MAX];
```
Comment (lemunozm) [inline src/c/core/serialization/xrce_protocol_internal.h]: Are we currently waste a `sizeof(Property)` size in the stack when a create message is created, right?

File: src/c/core/serialization/xrce_protocol_internal.h [hunk 2]
```diff
@@ -216,6 +216,7 @@ typedef struct CLIENT_Representation
     uint8_t session_id;
     bool optional_properties;
     PropertySeq properties;
+    uint16_t mtu;
```
Comment (lemunozm) [inline src/c/core/serialization/xrce_protocol_internal.h]: I don't see the serialization of `CLient_Representation` modified for this new attribute.

File: src/c/core/serialization/xrce_protocol_internal.h (hunk 1 above)
Comment (julianbermudez) [reply src/c/core/serialization/xrce_protocol_internal.h]: Yes, it wastes only a `sizeof(char*)` more than the previous one. Do you prefer to remove it?

File: src/c/core/serialization/xrce_protocol_internal.h (hunk 2 above)
Comment (julianbermudez) [reply src/c/core/serialization/xrce_protocol_internal.h]: You are right, sorry. I modified `CLIENT_Representation` de-serialization functions.

File: src/c/core/serialization/xrce_protocol_internal.h (hunk 1 above)
Comment (lemunozm) [reply src/c/core/serialization/xrce_protocol_internal.h]: It is ok for me, do not worry.

**Pull request merged**